base/div.jl

```diff
@@ -364,7 +364,66 @@ function div(x::T, y::T, ::typeof(RoundUp)) where T<:Integer
     return d + (((x > 0) == (y > 0)) & (d * y != x))
 end
 
-# Real
-# NOTE: C89 fmod() and x87 FPREM implicitly provide truncating float division,
-# so it is used here as the basis of float div().
-div(x::T, y::T, r::RoundingMode) where {T<:AbstractFloat} = convert(T, round((x - rem(x, y, r)) / y))
+function mw_div(x::NTuple{2,T}, y::T) where {T<:AbstractFloat}
+    (x_hi, x_lo) = x
+
+    # "DWDivFP3" AKA "Algorithm 15" from a paper by Joldes, Muller and
+    # Popescu: https://doi.org/10.1145/3121432
+    hi = x_hi / y
```

Comment:
I would be interested in if this speeds up and retains most of the accuracy if you replay `x_hi / y` with `x_hi * inv(y)` doing so would let you save a division on since then you could do the same for `lo = δ / y` (and since we only care about the high word of this, it could fold with the `canonicalize2` into `return fma(inv(y), δ, hi)` saving another instruction.